libraries/AP_HAL_SITL/sitl_barometer.cpp

```diff
@@ -76,7 +76,9 @@ void SITL_State::_update_barometer(float altitude)
 
     // find data corresponding to delayed time in buffer
     for (uint8_t i=0; i<=baro_buffer_length-1; i++) {
-        time_delta_baro = abs(delayed_time_baro - buffer_baro[i].time); // find difference between delayed time and time stamp in buffer
+        // find difference between delayed time and time stamp in buffer
+        time_delta_baro = abs(
+                (int32_t)(delayed_time_baro - buffer_baro[i].time));
```

Comment:
@rchiossi  I think that would happen if `_sitl->baro_delay` was extremely large, which wouldn't make any sense. Considering that `buffer_baro` is added a value to on every 10 msecs and that the maximum value of `baro_buffer_length` is 255 (currently 50), I don't believe we would have a problem here.